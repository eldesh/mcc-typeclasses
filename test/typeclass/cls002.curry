class A a where
  f :: a -> a
class A a => B a where
  g :: a -> a
  g = f

instance A Integer where
  f x = x
instance A Float where
  f _ = 0.0
instance A a => A (Maybe a) where
  f (Just x) = Just (f x)
  f Nothing = Nothing

-- NB The (A a) context is required by the super class instance
instance A a => B (Maybe a) where

main = (g (Just 1),g (Just 1.0))
