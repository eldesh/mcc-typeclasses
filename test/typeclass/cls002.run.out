(Just 1,Just 0.0)
